beaconChain/helpers/stateTransitionHelpers.ts

```diff
@@ -22,38 +23,72 @@ export function getActiveValidatorIndices(validators: ValidatorRecord[]): int[]
   }, []);
 }
 
+// Modified from: https://github.com/feross/buffer/blob/master/index.js#L1125
+function readUIntBE(array: Uint8Array, offset: number, byteLength: number): number {
+    let val: number = array[offset + --byteLength];
+    let mul: number = 1;
+    while (byteLength > 0) {
+        mul *= 0x100;
+        val += array[offset + --byteLength] * mul;
+    }
+    return val;
+}
+
 /**
  * The following is a function that shuffles any list; we primarily use it for the validator list.
  * @param {T[]} values
  * @param {hash32} seed
  * @returns {T[]} Returns the shuffled values with seed as entropy.
  */
-// TODO finish this
 function shuffle<T>(values: T[], seed: hash32): T[] {
   const valuesCount: int = values.length;
   // Entropy is consumed from the seed in 3-byte (24 bit) chunks.
-  const randBytes = 3;
+  const randBytes: number = 3;
   // Highest possible result of the RNG
-  const randMax = 2 ** (randBytes * 8) - 1;
+  const randMax: number = 2 ** (randBytes * 8) - 1;
 
   // The range of the RNG places an upper-bound on the size of the list that may be shuffled.
   // It is a logic error to supply an oversized list.
-  if (valuesCount < randMax) { throw new Error("Oversized list supplied to shuffle()!"); }
+  if (!(valuesCount < randMax)) { throw new Error("Oversized list supplied to shuffle!"); }
 
   // Make a copy of the values
   const output: T[] = values.slice();
```

Comment:
I don't think so. Do you think it could be an issue?

Reply:
Probably not. Its a generic function so it might be naive to expect anything of the inputs, but all of the objects we have defined so far are shallow.